game-core/src/main/java/games/strategy/engine/data/ModifiableProperty.java

```diff
@@ -0,0 +1,145 @@
+package games.strategy.engine.data;
+
+import java.util.function.Supplier;
+
+/**
+ * A wrapper interface to Bundle setters, getters and resetters of the same field.
+ *
+ * @param <T> The type of the field to set, get and reset.
+ */
+public interface ModifiableProperty<T> {
+
+  void setValue(String string) throws Exception;
+
+  void setValue(T value) throws Exception;
+
+  @SuppressWarnings("unchecked")
+  default void setObjectValue(final Object o) {
+    try {
+      if (o instanceof String) {
+        setValue((String) o);
+      } else {
+        setValue((T) o);
+      }
+    } catch (final Exception e) {
+      throw new IllegalStateException("Failed to set Attachment property", e);
```

Comment:
Not sure why this method wouldn't have the same exception specification as `setValue()`.  It seems to confuse things to throw a checked exception from some setters and an unchecked exception from others.  Would recommend being consistent with `setValue()` and just allow the checked exception to propagate.

Reply:
The reason I threw an unchecked exception here is because the calling methods do not throw any checked exceptions, feel free to try changing this, but I ran into way too much compile errors when trying quickly.